extension/experiments/voice/api.js

```diff
@@ -54,6 +58,19 @@ this.voice = class extends ExtensionAPI {
           async quitApplication() {
             return runCommand("cmd_quitApplication");
           },
+
+          async openSidebar() {
+            const sidebarAction = sidebarActionFor(extension);
+            const windowTracker = ChromeUtils.import(
+              "resource://gre/modules/Extension.jsm",
+              {}
+            ).Management.global.windowTracker;
+            /* https://searchfox.org/mozilla-central/source/browser/components/extensions/parent/ext-sidebarAction.js#495 */
+            const window = windowTracker.topWindow;
+            if (context.canAccessWindow(window)) {
+              sidebarAction.open(window);
+            }
```

Comment:
I was able to get this working for opening just the bookmarks sidebar with:

```js
          async openSidebar() {
            const windowTracker = ChromeUtils.import(
              "resource://gre/modules/Extension.jsm",
              {}
            ).Management.global.windowTracker;
            const window = windowTracker.topWindow;
            const SidebarUI = window.SidebarUI;
            await SidebarUI.show("viewBookmarksSidebar");
          },
```

Presumably History and Synced Tabs have other ids besides `"viewBookmarksSidebar"`, but I didn't look them up on searchfox.org. When hiding the sidebar, `SidebarUI.hide()` should work with no argument.

Reply:
Hi @ianb,
as I was working on this, I was thinking why do not we create a sidebar specifically for the extension that would show all the **intents used by the user**.

I know there is a "history" tab in settings.

But getting a quick view of the used intents in a toggle-able i**ntent-sidebar** would also be nice.
What is your thoughts on this?

Reply:
I left them there thinking of developing them to be the intent-sidebar.
I will ask to @awallin, what he thinks of this feature.
Thanks lots for all your guidance :)